server/external/rapidapiClient.test.js

```diff
@@ -2,10 +2,45 @@ const axios = require('axios')
 const RapidapiClient = require('./rapidapiClient')
 
 jest.mock('axios');
+jest.mock('../redis.repository');
+
+const { checkCache, getCachedCall, cacheData } = require('../redis.repository');
+
+afterEach(() => {
+    jest.clearAllMocks();
+});
+
+describe('API caching tests', () => {
+    test('should fetch from API when URL not in cache', async () => {
+        console.log('testing not in cache');
+        checkCache.mockImplementation(() => 0);
+        cacheData.mockImplementation(() => "OK");
+
+        const resp = {data:"123"};
+        axios.request.mockResolvedValue(resp);
+
+        const result = await RapidapiClient.makeQuery('hello', {hi:'abc'});
+        expect(result).toEqual('123');
+        
+    });
+
+    test('should fetch from redis cache when URL is present', async () => {
+        console.log('testing in cache');
+        checkCache.mockImplementation(() => 1);
+
```

Comment:
Good job mocking for the unit tests!

Reply:
Thanks! These mocks really help with testing the functionality of the rapidapiClient code logic in isolation, and making sure that dependencies/setup are not resulting in tests that accidentally fail

Reply:
I see. I'll take a look at this next time I write unit tests for Redis. Sounds useful.